### ISSUE TITLE
System is freezing after last update

### ISSUE THREAD

**Issue by PrzemekSkw** [pop-os/pop#569]
**Distribution (run `cat /etc/os-release`):**
NAME="Pop!_OS"
VERSION="19.04"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 19.04"
VERSION_ID="19.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=disco
UBUNTU_CODENAME=disco

Laptop Dell with nvidia graphics
Intel® Core™ i5-7300HQ CPU @ 2.50GHz × 4 
nvidia driver 418.56
RAM: 16 GB
Gnome 3.32.1

**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
any

**Issue/Bug Description:**
System is freezeing all the time for few seconds.
Watching movies, playing games, switch desktops, useing email client or web browser.

**Comment by wmutschl** [comment]
I have the exact same issue on my Workstation-Laptop Dell Precision 7520 with current nvidia graphics (as provided by Pop-OS). I now switched to Intel graphics to see whether this is related to nvidia...

**Comment by wmutschl** [comment]
Okay, so far no more freezing with intel... issue seems to be nvidia related. Could you also try to switch to intel graphics (https://support.system76.com/articles/graphics-switch-pop/) and confirm whether this also "solves" your issue?

**Comment by PrzemekSkw** [comment]
Yes, I upgrade Kernel to 5.1 and that helps but don't have nvidia drivers and can't find any.

**Comment by wmutschl** [comment]
I did not upgrade the Kernel, my kernel is
```
wmutschl@pop-os:~$ uname -a
Linux pop-os 5.0.0-15-generic #16pop0-Ubuntu SMP Wed May 15 14:41:25 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```
I simply turned off the nvidia card and switched to my intel card as is described in this link: https://support.system76.com/articles/graphics-switch-pop/.
The following command confirms that I am now using my Intel Graphics card instead of my Nvidia card.
```
wmutschl@pop-os:~$ lspci -vnnn | perl -lne 'print if /^\d+\:.+(\[\S+\:\S+\])/' | grep VGA
00:02.0 VGA compatible controller [0300]: Intel Corporation HD Graphics 630 [8086:591b] (rev 04) (prog-if 00 [VGA controller])
```

**Comment by PrzemekSkw** [comment]
@wmutschl thanks, but that also don't fix that problem. I have gaming laptop and I want to play Dark Souls :D

**Comment by wmutschl** [comment]
Sure I understand that ;-)
Maybe updating to the latest nvidia driver from the official nvidia ppa will solve your issue (if it is really the graphics driver), e.g. as outlined here: http://www.linuxandubuntu.com/home/how-to-install-latest-nvidia-drivers-in-linux
As I don't need the nvidia card right now, I will just stay with the intel card for now.

**Comment by PrzemekSkw** [comment]
OK, I downgrade kernel to 5.0 and install last nvidia driver 430.15. Don't freezeing now :)

**Comment by Wilkolicious** [comment]
Also having this problem since running an upgrade late last week.

XPS15 (9560 I think)
![image](https://user-images.githubusercontent.com/16191979/58182128-f3e71800-7ca4-11e9-9a9f-52654714fa9f.png)

```
❯ cat /etc/os-release
NAME="Pop!_OS"
VERSION="19.04"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 19.04"
VERSION_ID="19.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=disco
UBUNTU_CODENAME=disco
```

```
❯ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Pop!_OS 19.04
Release:	19.04
Codename:	disco
```

```
❯ uname -srvi
Linux 5.0.0-15-generic #16pop0-Ubuntu SMP Wed May 15 14:41:25 UTC 2019 x86_64
```

```
❯ modinfo nvidia | grep version
version:        418.56
srcversion:     AA5D9C867B6BC7FFD055995
```

Reproducible every time by dragging tabs in vscode.

**Comment by Nebrethar** [comment]
Just got a new System76 oryp5 and it has this issue on Pop_OS. 
I have the issue on both Nvidia and Intel graphics.
This is happening every few minutes or so of regular use.
If i use compatibility software (wine or proton) for **anything**, the laptop becomes frozen every few seconds of use, for about 20 seconds each time.

The system performs as expected otherwise.
```
-> cat /etc/os-release
NAME="Pop!_OS"
VERSION="19.04"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Pop!_OS 19.04"
VERSION_ID="19.04"
HOME_URL="https://system76.com/pop"
SUPPORT_URL="http://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=disco
UBUNTU_CODENAME=disco
```
```
-> lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Pop!_OS 19.04
Release:	19.04
Codename:	disco
```
```
-> uname -srvi
Linux 5.0.0-13-generic #14-Ubuntu SMP Mon Apr 15 14:59:14 UTC 2019 x86_64
```
```
-> nvidia-smi
| NVIDIA-SMI 418.56       Driver Version: 418.56       CUDA Version: 10.1     |
|   0  GeForce RTX 207...  Off  | 00000000:01:00.0 Off |                  N/A |
| N/A   41C    P8     7W /  N/A |    351MiB /  7952MiB |     20%      Default |
```
![Screenshot from 2019-05-22 17-53-18](https://user-images.githubusercontent.com/34951354/58214205-89d96d80-7cba-11e9-9425-b6bc740f5527.png)

**Comment by BaronKrause** [comment]
I have a Lenovo X1 Extreme and it also does this ever since the Nvidia update that fixed the Screen tearing/Prime Display Sync issue.

You can even hear the fans spin up when it freezes intermittently.

It does not happen when set to Intel, and it also does not happen when set to Nvidia in the OS and set to Nvidia only (Discrete mode) in the bios. So it isn't just an Nvidia issue, but an issue with the Nvidia/Intel display syncing system.

**Comment by PrzemekSkw** [comment]
@BaronKrause  @Nebrethar As I said above upgrade nvidia drivers to 430.15 helps for me and for @Wilkolicious.

**Comment by BaronKrause** [comment]
When a driver is installed from a third party repo, will it have any of the tweaks that the pop team did to fix the screen tearing/prime monitor issues?

Curious if the fix from 430 is because of driver 430 or just because your not using the pop supplied driver/tweaks.

**Comment by PrzemekSkw** [comment]
@BaronKrause I don't know. It works and games that I play work great so it's fine for me.

**Comment by wmutschl** [comment]
> @PrzemekSkw 
> Curious if the fix from 430 is because of driver 430 or just because your not using the pop supplied driver/tweaks.

That is a good question. Does anyone have a link to a blog article or github issue where the developer of PopOS explain how they do things different with nvidia than the "ppa:graphics-drivers"? I am aware that PopOS ships with newer drivers than stock Ubuntu (and I very much like that), but as @PrzemekSkw  was saying, what else are they doing in terms of tweaks and optimizations?

**Comment by wmutschl** [comment]
Thanks for the clarification. I did have some screen tearing in firefox with the latest driver from the ppa "ppa:graphics-drivers", so I applied `nvidia-drm.modeset=1` manually and now everything works smooth. 

For reference, I followed this [blog](https://www.cmscritic.com/stop-screen-tearing-with-optimus-laptops-using-nvidia-drivers-in-linux/) or this [forum post](https://ubuntuforums.org/showthread.php?t=2374405).

First, I created a file:
`sudo nano /etc/modprobe.d/zz-nvidia-modeset.conf`
and added the following line into this file:
`options nvidia_drm modeset=1`
Save and exit nano, and run
`sudo update-initramfs -u`
Reboot. After the reboot, check whether
`sudo cat /sys/module/nvidia_drm/parameters/modeset`
outputs a `Y`. For me this solved the screen tearing with the latest nvidia driver on my Dell Precision 7520 Notebook with Pop!_OS 19.04.
